src/demands/demands.controller.ts

```diff
@@ -11,13 +11,32 @@ import { UserRole } from '../users/types/user-role.enum';
 import { AuthGuard } from '../guards/authentication.guard';
 import { UpdateDemandDto } from './dto/updateDemandDto';
 import { errorHandler } from '../error/error-mapper';
+import NotFoundError from '../error/not-found.error';
 
 @ApiTags('demands')
 @Controller('demands')
 @Injectable()
 export class DemandsController {
   constructor(private readonly sequelize: Sequelize, private readonly demandsService: DemandsService) {}
 
+  @ApiResponse({ isArray: true, type: Demand, description: 'returns single demand' })
+  @Get('/:id')
+  public async getDemand(@Param('id') id: string): Promise<Demand | void> {
+    try {
+      return await this.sequelize.transaction(async (transaction) => {
+        const demand = await this.demandsService.getDemandById(transaction, id);
+
+        if (!demand) {
+          throw new NotFoundError(`Demand ${id} not found`);
```

Comment:
Imo it is better to return it as an "enum like" error
```suggestion
          throw new NotFoundError(`DEMAND_NOT_FOUND`);
```